lib/rubocop/cop/layout/space_around_method_call.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Layout
+      #   Checks that there are no spaces surrounding a dot method call.
+      #   # bad
+      #   object .some_method
+      #
+      #   # bad
+      #   object. some_method
+      #
+      #   # bad
+      #   object
+      #    .some_method .another_method
+      #
+      #   # good
+      #   object.some_method
+      #
+      #   # good
+      #   object
+      #    .some_method
+      #
+      class SpaceAroundMethodCall < Cop
+        include SurroundingSpace
+
+        AFTER_MSG = 'Avoid using space after dot in method call.'
+        BEFORE_MSG = 'Avoid using space before dot in method call.'
+
+        def on_send(node)
+          tokens = tokens(node)
+
+          left = left_side_of_dot(tokens)
+          right = right_side_of_dot(tokens)
+          dot = dot_token(tokens)
+          return unless dot
+
+          no_space_offenses(node, dot, right, AFTER_MSG)
+          return if left.line != dot.line
+
+          no_space_offenses(node, left, dot, BEFORE_MSG)
+        end
+
+        private
+
+        def dot_token(tokens)
+          tokens.reverse.find(&:dot?)
```

Comment:
This will find the wrong dot (the second one) in cases like `foo . bar(foo.bar)` and hence not detect the spaces around the first one.

Reply:
The reason I did `reverse` was because when doing `foo.bar.foo2 . bar2` on the first `on_send` the `node` contains `foo.bar.foo2 . bar2` on the second `on_send` the node contains just `foo.bar.foo2` so I needed to get the last token to find the space there. 

You are correct that with `foo . bar(foo.bar)` using the `reverse` will not detect the space. 

Any ideas on how to get both scenarios to work?

Reply:
You could make use of the source map of the send node in order to find the right dot: `send_node.loc.dot` will give you the location of the dot in the source which you could compare to the token positions.

Be aware that `loc.dot` can be nil because there are send nodes without dots in the source. You might want to cover this case in your specs. There is a convenience method `dot?` for send nodes, so you could do something like this:
```ruby
def on_send(node)
  return unless node.dot?
```